retos/entregas/ikerCelaya/whacAMole/whacAMole.java

```diff
@@ -0,0 +1,56 @@
+import java.util.Scanner;
+
+public class whacAMole {
+    public static void main(String[]args){
+        int dimension = 4;
+        int turno = 1;
+        int acierto = 0;
+        int casilla = 0;
+        int topo = 0; 
+        int topo1=0;
+        int contador = 1;
+
+        Scanner entrada =  new Scanner(System.in);
+
+        do{
+            contador = 1;
+            topo = ((int) (Math.random() * 100) % 16) + 1;
+            do{
+                topo1 = ((int) (Math.random() * 100) % 16) + 1;
+            }
+            while(topo==topo1);
+            do{
+                System.out.println("Golpe (entre 1 y 16)");
+                casilla = entrada.nextInt();
+            } 
+            while(casilla < 0 || casilla > 16);
+            for(int j = 1; j <= dimension; j = j + 1) {
+                for(int i = 1; i <= dimension; i = i + 1) {
+                    if(casilla == topo && casilla == contador) {
+                        System.out.print(" [**] ");
+                        acierto++;
+                    } 
+                    else if(casilla == topo1 && casilla == contador) {
+                        System.out.print(" [**] ");
+                        acierto++;
+                    } 
+                    else if(contador == casilla) {
+                        System.out.print(" [[]] ");
+                    } 
+                    else if(contador == topo || contador==topo1) {
+                        System.out.print(" [''] ");
+                    } 
+                    else {
+                        System.out.print(" (  ) ");
+                    }
+                    contador++;
+                }
+                System.out.println();
+            }
```

Comment:
Mira esta propuesta de código, con un solo for, y usando println cuando corresponde (o sea, cuando llega al elemento múltiplo de dimension, que para el ejemplo sería 7, 14, 21 y 28)
```
String queImprimir;
int dimension=7;
int numeroFilas=4;

for (int contador=0; contador<=dimension*numeroFilas; contador++) {
  queImprimir = " (  ) ";  
  if ((contador==topo1 || contador==topo2) && (casilla==contador)) {
    queImprimir = " [**] ";
    acierto++;
  } else if (casilla==contador) {
    queImprimir = " [[]] ";
  } else if (contador==topo1 || contador==topo2) {
    queImprimir = " [''] ";
  } 
  System.out.print(queImprimir);
  if (contador%dimension=0) {
    System.out.println();
  }
}
```

Reply:
¡Perdón! me equivoqué en la comparación del if y he usado ```=``` en lugar de ```==``` Lo corrijo en github...